File: python-algorithm/algo_03_two_points/L344.cpp
//
// Created by 李恒 on 2020/1/3.
//

// 344 反转字符串

#include <iostream>
#include <cstring>
#include <vector>

using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        for(int i=0 ;i < s.size()/2; i++)
            swap(s[i],s[s.size()-1-i]);
    }
};

File: python-algorithm/algo_03_two_points/L344.py
#--coding:utf-8--


class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        lt, rt = 0, len(s)-1
        while lt < rt:
            s[lt], s[rt] = s[rt], s[lt]
            lt += 1
            rt -= 1

